### ISSUE TITLE
compat entry for Test

### ISSUE THREAD

**Issue by putianyi889** [JuliaCI/PkgTemplates.jl#466]
The auto-generated Project.toml doesn't pass Aqua test:
```
PkgName [uuid] does not declare a compat entry for the following extras:
1-element Vector{Base.PkgId}:
 Test [8dfed614-e22c-5e08-85e1-65c5234f0b40]
```

**Comment by gdalle** [comment]
Good catch! Do you want to try and fix it in a PR? I can help

**Comment by putianyi889** [comment]
> Good catch! Do you want to try and fix it in a PR? I can help

I'd like to, but I'm not sure if I'll need to change the definition of `Template` as it stores the Julia version.

**Comment by oxinabox** [comment]
I think when you enable Aqua as a test argument in it also fails for Aqua itself

**Comment by gdalle** [comment]
No, I think it just requires editing the `Test` plugin. Two cases to consider: separate `Project.toml` file or `[extras]` section in the main one. I think in both cases, you should be able to set the version for `Test` with `Pkg.compat`: https://pkgdocs.julialang.org/v1/api/#Pkg.compat

The relevant lines to edit are here:

https://github.com/JuliaCI/PkgTemplates.jl/blob/805bb7cfe158ab92ba76af3f2af5d7225f150714/src/plugins/tests.jl#L102-L153
